src/index.ts

```diff
@@ -1,76 +1,84 @@
-/*
-    This file is an example logical flow from a URL to
-    fetching and parsing repo data and calculating some metrics
-*/
-
-import 'dotenv/config';
-import { fetchRepoData } from "./api/githubApi";
+import { Worker } from 'worker_threads';
+import { fetchRepoData } from './api/githubApi';
 import { getRepoDetails } from './utils/urlHandler';
 import { initLogFile, logToFile, metricsLogToStdout } from './utils/log';
-import { Worker } from 'worker_threads';
-import { calculateMetrics } from './metricCalcs';
+import { calculateMetrics } from './metrics/metric';
 
+// CommonJS-style import for dotenv
+const dotenv = require('dotenv');
+
+// Load environment variables from .env file
+dotenv.config();
 
 // Function to create and manage worker threads
-export function runWorker(owner, repo, token, repoURL, repoData, metric) {
+export function runWorker(
```

Comment:
Good organization, format of parameters will need to change to align with eslint syntax